named_entity_recognition/named_entity_recognition_question_answering.ipynb

```diff
@@ -0,0 +1,1341 @@
+    "\n",
+    "annotations = [i['instances'] for i in sa_response]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "58d74089",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prompt_mapper = {'CASE_NUMBER': 'case number',\n",
+    " 'COURT': 'court',\n",
+    " 'DATE':'date',\n",
+    " 'GPE':'location',\n",
+    " 'JUDGE':'judge',\n",
+    " 'LAWYER':'lawyer',\n",
+    " 'ORG':'organization',\n",
+    " 'OTHER_PERSON':'other person',\n",
+    " 'PETITIONER':'petitioner',\n",
+    " 'PRECEDENT':'precedent',\n",
+    " 'PROVISION':'provision',\n",
+    " 'RESPONDENT':'respondent',\n",
+    " 'STATUTE':'statute',\n",
+    " 'WITNESS':'witness'}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8149d73e",
+   "metadata": {},
+   "source": [
+    "## 3. Prepare data for Bert NER model\n",
+    "<a id='prepare_data_for_bert_model'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e7125cf-7d46-4ff6-b111-35112c645146",
+   "metadata": {},
+   "source": [
+    "We will use pretrained tokenizer *bert-base-cased* for our data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "142f392e-e19a-45c5-a458-d3f9003b78ee",
+   "metadata": {},
+   "source": [
+    "Now we can create class for our dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "789fc352",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Instance = namedtuple(\"Instance\", [\"context\", \"question\", \"answer\"])\n",
+    "\n",
+    "class Dataset(torch.utils.data.Dataset):\n",
+    "    def __init__(self,\n",
+    "                 qa_sentences,\n",
+    "                 qa_labels,\n",
+    "                 prompt_mapper):\n",
+    "        super().__init__()\n",
+    "        self.prompt_mapper = prompt_mapper\n",
+    "        self.dataset = self._prepare_dataset(qa_sentences=qa_sentences,\n",
+    "                                             qa_labels=qa_labels)\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.dataset)\n",
+    "\n",
+    "    def __getitem__(self, idx: int):\n",
+    "        return self.dataset[idx]\n",
+    "\n",
+    "    def _prepare_dataset(self,\n",
+    "                         qa_sentences,\n",
+    "                         qa_labels):\n",
+    "\n",
+    "        dataset = []\n",
+    "        for sentence, labels in tqdm(zip(qa_sentences, qa_labels),desc=\"prepare_dataset\"):\n",
+    "            for label_tag, label_name in self.prompt_mapper.items():\n",
+    "                question_prompt = f\"What is the {label_name}?\"\n",
+    "\n",
+    "                answer_list = []\n",
+    "                for span in labels:\n",
+    "                    if span['label'] == label_tag:\n",
+    "                        answer_list.append(span)\n",
+    "\n",
+    "                if len(answer_list) == 0:\n",
+    "                    empty_span = {\"token\" : \"\",\n",
```

Comment:
What is SPAN? What is Prompt? What is Instance? 
Add more text about terminology and a few words on how it's working